testsuite/tests/warnings/w53.ml

```diff
@@ -1,57 +1,507 @@
+ setup-ocamlc.byte-build-env;
+ compile_only = "true";
+ ocamlc.byte;
+ check-ocamlc.byte-output;
+*)
 
+module type TestAlertSig = sig
+  type t1 = Foo1 [@alert foo "foo"] (* accepted *)
 
+  val x : int [@alert foo "foo"] (* rejected *)
 
+  type 'a t2 = 'a [@@alert foo "foo"] (* accepted *)
 
+  val y : int [@@alert foo "foo"] (* accepted *)
 
+  [@@@alert foo "foo"] (* rejected *)
+end
 
+module TestAlertStruct = struct
+  let x = 5 [@alert foo "foo"] (* rejected *)
 
-*)
+  let y = 10 [@@alert foo "foo"] (* rejected *)
 
-let h x = x [@inline] (* rejected *)
-let h x = x [@ocaml.inline] (* rejected *)
+  [@@@alert foo "foo"] (* rejected *)
+end
 
-let i x = x [@inlined] (* rejected *)
-let j x = x [@ocaml.inlined] (* rejected *)
-let k x = (h [@inlined]) x (* accepted *)
-let k' x = (h [@ocaml.inlined]) x (* accepted *)
-let l x = h x [@inlined] (* rejected *)
 
-let m x = x [@tailcall] (* rejected *)
-let n x = x [@ocaml.tailcall] (* rejected *)
-let o x = (h [@tailcall]) x (* accepted *)
-let p x = (h [@ocaml.tailcall]) x (* accepted *)
-let q x = h x [@tailcall] (* rejected *)
+module type TestBoxedSig = sig
+  type t1 = { x : int [@boxed] } (* rejected *)
 
-module type E = sig end
+  val x : int [@boxed] (* rejected *)
 
-module A(E:E) = struct end [@@inline] (* accepted *)
-module A'(E:E) = struct end [@@ocaml.inline] (* accepted *)
-module B = ((functor (E:E) -> struct end) [@inline]) (* accepted *)
-module B' = ((functor (E:E) -> struct end) [@ocaml.inline]) (* accepted *)
-module C = struct end [@@inline] (* rejected *)
-module C' = struct end [@@ocaml.inline] (* rejected *)
-module D = struct end [@@inlined] (* rejected *)
-module D' = struct end [@@ocaml.inlined] (* rejected *)
+  type t2 = { x : int } [@@boxed] (* accepted *)
 
-module F = (A [@inlined])(struct end) (* accepted *)
-module F' = (A [@ocaml.inlined])(struct end) (* accepted *)
-module G = (A [@inline])(struct end) (* rejected *)
-module G' = (A [@ocaml.inline])(struct end) (* rejected *)
+  val y : int [@@boxed] (* rejected *)
 
-module H = Set.Make [@inlined] (Int32) (* GPR#1808 *)
+  [@@@boxed] (* rejected *)
+end
 
-module I = Set.Make [@inlined]
-module I' = Set.Make [@ocaml.inlined]
+module TestBoxedStruct = struct
+  let x = (42 [@boxed], 84) (* rejected *)
 
-module J = Set.Make [@@inlined]
-module J' = Set.Make [@@ocaml.inlined]
+  let y = 10 [@@boxed] (* rejected *)
 
-(* TEST
- flags = "-w +A-60-70";
- setup-ocamlc.byte-build-env;
- compile_only = "true";
- ocamlc.byte;
- check-ocamlc.byte-output;
-*)
+  [@@@boxed] (* rejected *)
+end
+
+
+module type TestDeprecatedSig = sig
+  type t1 = Foo1 [@deprecated] (* accepted *)
+
+  val x : int [@deprecated] (* rejected *)
+
+  type 'a t2 = 'a [@@deprecated] (* accepted *)
+
+  val y : int [@@deprecated] (* accepted *)
+
+  [@@@deprecated] (* rejected *)
+end
+
+module TestDeprecatedStruct = struct  (* CJC XXX THIS IS ALL BUGGY *)
+  let x = 5 [@deprecated] (* rejected *)
```

Comment:
What is buggy in this test?

Reply:
Oops, sorry, nice catch!  This was a stray comment from an intermediate state, now deleted.